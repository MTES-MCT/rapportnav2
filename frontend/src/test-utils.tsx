import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
