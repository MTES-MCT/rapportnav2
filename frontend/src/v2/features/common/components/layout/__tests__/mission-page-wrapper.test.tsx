import { render, screen } from '../../../../../../test-utils'
import MissionPageWrapper from '../mission-page-wrapper'

describe('MissionPageWrapper', () => {
  it('should match the snapshot', () => {
    const wrapper = render(
      <MissionPageWrapper
        missionFooter={<>Footer</>}
        missionHeader={<>Header</>}
        missionAction={<>Action</>}
        missionTimeLine={<>Timeline</>}
        missionGeneralInformations={<>Général informations</>}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should show loading message', () => {
    render(<MissionPageWrapper isLoading={true} missionFooter={<>Footer</>} missionHeader={<>Header</>} />)
    expect(screen.getByText('Chargement...')).toBeInTheDocument()
  })

  it('should show error message', () => {
    render(
      <MissionPageWrapper
        missionFooter={<>Footer</>}
        missionHeader={<>Header</>}
        error={{ message: 'My beautiful error' }}
      />
    )
    expect(screen.getByText('My beautiful error')).toBeInTheDocument()
  })
})
