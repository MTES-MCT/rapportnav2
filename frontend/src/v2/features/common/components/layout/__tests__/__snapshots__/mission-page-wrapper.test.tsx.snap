// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MissionPageWrapper > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin: 0px; display: flex; min-height: 100vh; max-height: 100vh; flex-direction: column;"
      >
        Header
        <div
          class="rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-space-between"
          style="display: flex; flex: 1;"
        >
          <div
            class="rs-flex-box-grid-item rs-flex-box-grid-item-8"
            style="flex: 1; padding: 2rem; height: 2000px; overflow-y: hidden; min-height: calc(100vh - 2 * 60px); max-height: calc(100vh - 2 * 60px);"
          >
            Général informations
          </div>
          <div
            class="rs-flex-box-grid-item rs-flex-box-grid-item-8"
            style="flex: 1; padding: 2rem; height: 2000px; overflow-y: hidden; min-height: calc(100vh - 2 * 60px); max-height: calc(100vh - 2 * 60px); background-color: rgb(247, 247, 250);"
          >
            Timeline
          </div>
          <div
            class="rs-flex-box-grid-item rs-flex-box-grid-item-8"
            style="flex: 1; padding: 2rem; height: 2000px; overflow-y: hidden; min-height: calc(100vh - 2 * 60px); max-height: calc(100vh - 2 * 60px); background-color: rgb(229, 229, 235);"
          >
            Action
          </div>
        </div>
        Footer
      </div>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <div
      style="margin: 0px; display: flex; min-height: 100vh; max-height: 100vh; flex-direction: column;"
    >
      Header
      <div
        class="rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-space-between"
        style="display: flex; flex: 1;"
      >
        <div
          class="rs-flex-box-grid-item rs-flex-box-grid-item-8"
          style="flex: 1; padding: 2rem; height: 2000px; overflow-y: hidden; min-height: calc(100vh - 2 * 60px); max-height: calc(100vh - 2 * 60px);"
        >
          Général informations
        </div>
        <div
          class="rs-flex-box-grid-item rs-flex-box-grid-item-8"
          style="flex: 1; padding: 2rem; height: 2000px; overflow-y: hidden; min-height: calc(100vh - 2 * 60px); max-height: calc(100vh - 2 * 60px); background-color: rgb(247, 247, 250);"
        >
          Timeline
        </div>
        <div
          class="rs-flex-box-grid-item rs-flex-box-grid-item-8"
          style="flex: 1; padding: 2rem; height: 2000px; overflow-y: hidden; min-height: calc(100vh - 2 * 60px); max-height: calc(100vh - 2 * 60px); background-color: rgb(229, 229, 235);"
        >
          Action
        </div>
      </div>
      Footer
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
